students/jerickson/Lesson6/test_mailroom.py

```diff
@@ -0,0 +1,542 @@
+"""Unit Tests for rfb_station.scale_reader module
+
+Test Cases:
+    Positive: Case that runs without errors
+    Negative: Case that raises unhandled errors
+"""
+
+# pylint: disable=redefined-outer-name
+# pylint: disable=invalid-name
+# pylint: disable=no-self-use
+# pylint: disable=too-few-public-methods
+# pylint: disable=too-many-locals
+# pylint: disable=too-many-arguments
+
+import pytest
+from pytest_mock import mocker  # pylint: disable=unused-import
+
+import mailroom
+
+
+class Test_Report_CLI:
+    """
+    Tests the mailroom.menu_selection function.
+
+    __builtins__.print() is mocked to simulate user-interaction
+    mailroom.report() is mocked to provide an isolated state for each test
+    """
+
+    @pytest.mark.parametrize(
+        "list_len",
+        [
+            pytest.param(0, id="empty_report"),
+            pytest.param(1, id="one_line"),
+            pytest.param(10, id="ten_lines"),
+        ],
+    )
+    def test_report_cli(self, mocker, list_len):
+        """Positive-Test-Cases"""
+        # Setup
+        report_list = range(1, list_len + 1)
+
+        # Mock
+        mocked_print = mocker.patch.object(mailroom, "print")
+        mocker.patch.object(mailroom, "report", return_value=report_list)
+
+        # Execute
+        mailroom.report_cli()
+
+        # Assert
+        for _call in mocked_print.call_args_list:  # Report printed in order
+            _call_arg = _call.args[0]  # Argument print was called with
+            assert _call_arg == report_list[_call_arg - 1]
+
+
+class Test_Report:
```

Comment:
there is no reason to use class with a single test function (classes are used with `unittest` module)

with `pytest` you can just use functions directly.

Reply:
I use them like that for consistency so that looking through the code is a bit more clear. Is this bad practice?
Thank you for the feedback!!

Reply:
yes it is not standard way to write pytest tests. they are usually written as individual functions.